docs/whatsnew/7.0.rst

```diff
@@ -288,6 +289,29 @@ New ``Lorentz2D`` model
 A new 2D Lorentzian model has been added to the ``astropy.modeling``
 package.
 
+.. _whatsnew-7.0-masked-coordinates:
+
+Support for masks in ``coordinates``
+====================================
+
+Initial support for masks has been implemented for |SkyCoord|, as well as for
+coordinate frames and coordinate representations.
+
+Like more generally for |Masked| instances, one can access the mask of using
+``.mask``, check whether the underlying data are masked using ``masked``,
```

Comment:
I tried that, but it is not possible: `SkyCoord` does not show inherited members in the documentation. This is of course a general problem, in particular with the fact that `SkyCoord` uses `__getattr__` to have also all the attributes of the underlying frame. And I'd prefer not to "solve" this by referring to the `CoordinateMixin`...

Reply:
The main motivation of introducing `CoordinateSharedMethods` is to avoid code duplication, which is convenient for developers. However, if having the methods there hides their documentation from the users then that is inconvenient for them, and in public API user convenience is more important than developer convenience.

Reply:
It turns out removing `CoordinateSharedMethods` means all 3 still need to be implemented in `SkyCoord`: `get_mask` because of the attributes, `mask` and `masked` because they are abstract properties that any actual class needs to implement. For now, I've kept this in a separate commit.

Reply:
Done.

Reply:
Why doesn't SkyCoord show inherited methods?

Reply:
I actually do not know - but I remember trying it briefly and the docs build failed. Didn't quite have time to investigate further.

Reply:
Ok. We should revisit this. Because I do think large classes like SkyCoord can benefit from being refactored to inherit from a set of **completely orthogonal** base classes.